crates/biome_js_analyze/src/analyzers/style/use_consistent_array_type.rs

```diff
@@ -0,0 +1,485 @@
+                } else if let AnyTsType::TsArrayType(array_type) = ty {
+                    let ele_type = array_type.element_type().ok();
+                    Some(generate_array_type(ele_type, false))
+                } else {
+                    None
+                }
+            } else {
+                Some(element_type)
+            };
+
+            generate_array_type(element_type, true)
+        }
+    })
+}
+
+fn get_diagnostic_title<'a>(array_kind: TsArrayKind) -> Markup<'a> {
+    match array_kind {
+        TsArrayKind::GenericArray => {
+            markup! {"Use "<Emphasis>"shorthand T[] syntax"</Emphasis>" instead of "<Emphasis>"Array<T> syntax."</Emphasis>}
+        }
+        TsArrayKind::ReadonlyGenericArray => {
+            markup! {"Use "<Emphasis>"shorthand readonly T[] syntax"</Emphasis>" instead of "<Emphasis>"ReadonlyArray<T> syntax."</Emphasis>}
+        }
+        TsArrayKind::Simple => {
+            markup! {"Use "<Emphasis>"Array<T> syntax"</Emphasis>" instead of "<Emphasis>"shorthand T[] syntax."</Emphasis>}
+        }
+        TsArrayKind::Readonly => {
+            markup! {"Use "<Emphasis>"ReadonlyArray<T> syntax"</Emphasis>" instead of "<Emphasis>"shorthand readonly T[] syntax."</Emphasis>}
+        }
+    }
+}
+
+fn get_action_message(array_kind: TsArrayKind) -> MarkupBuf {
+    match array_kind {
+        TsArrayKind::GenericArray => {
+            markup! { "Use "<Emphasis>"shorthand T[] syntax"</Emphasis>" to replace" }.to_owned()
+        }
+        TsArrayKind::ReadonlyGenericArray => {
+            markup! { "Use "<Emphasis>"shorthand readonly T[] syntax"</Emphasis>" to replace" }
+                .to_owned()
+        }
+        TsArrayKind::Simple => {
+            markup! { "Use "<Emphasis>"Array<T> syntax"</Emphasis>" to replace"}.to_owned()
+        }
+        TsArrayKind::Readonly => {
+            markup! { "Use "<Emphasis>"ReadonlyArray<T> syntax"</Emphasis>" to replace"}.to_owned()
+        }
+    }
+}
+
+fn generate_array_type<I>(element_type: I, is_readonly: bool) -> AnyTsType
+where
+    I: IntoIterator<Item = AnyTsType>,
+    I::IntoIter: ExactSizeIterator,
+{
+    let ident = if is_readonly {
+        make::ident("ReadonlyArray")
+    } else {
+        make::ident("Array")
+    };
+
+    AnyTsType::TsReferenceType(
+        make::ts_reference_type(AnyTsName::JsReferenceIdentifier(
+            make::js_reference_identifier(ident),
+        ))
+        .with_type_arguments(make::ts_type_arguments(
+            make::token(T![<]),
+            make::ts_type_argument_list(element_type, None),
+            make::token(T![>]),
+        ))
+        .build(),
+    )
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct ConsistentArrayType {
+    /// "simple" | "generic"
+    pub consistent_array_type: &'static str,
+}
+
+impl Default for ConsistentArrayType {
+    fn default() -> Self {
+        Self {
+            consistent_array_type: "simple",
+        }
+    }
+}
```

Comment:
I suggest renaming the type to `ConsistentArrayTypeOptions` to match naming convention of other rules' options. We could also introduce an `enum` for the array type.

```suggestion
#[derive(Deserialize, Serialize, Default, Debug, Clone, Eq, PartialEq, Bpaf)]
#[cfg_attr(feature = "schemars", derive(JsonSchema))]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct ConsistentArrayTypeOptions {
    #[bpaf(hide)]
    syntax: ConsistentArrayType,
}

#[derive(Deserialize, Serialize, Debug, Default, Clone, Copy, Eq, PartialEq)]
#[cfg_attr(feature = "schemars", derive(JsonSchema))]
#[serde(rename_all = "camelCase")]
pub enum ConsistentArrayType {
    /// `ItemType[]`
    #[default]
    Shothand,
    /// `Array<ItemType>`
    Generic,
}
```

Also, you have to derive `Bpaf` and implement `FromStr` trait for these two types.
You must also implement `Deserializable` for these two types.
You can take a look to other rules that have options, such as [useNamingConvention](https://github.com/biomejs/biome/blob/main/crates/biome_js_analyze/src/semantic_analyzers/style/use_naming_convention.rs#L510-L614).

Reply:
Done~